[WIP] CAM-12

dfrojas opened the pull request:
# CAM 12

## Description

This PR adds a modal displaying information about sensitive data to download. This modal is displayed when the user does click on the Data Download tab.

## How to testing

You need to add the flag `ENABLE_SENSITIVE_DATA_MSG_ON_DASHBOARD` in settings or in site_configuration

## Screenshots

![screenshot from 2018-01-26 11-36-06](https://user-images.githubusercontent.com/3374009/35450207-48081d32-028d-11e8-8ad0-fb3d053b55f3.png)

![screenshot from 2018-01-26 11-36-21](https://user-images.githubusercontent.com/3374009/35450233-5da3ddca-028d-11e8-973f-ed43cb2820c6.png)




## Reviewers

- [ ] @felipemontoya 

## Post-review

- [ ] Squash the commits

dfrojas commented on lms/templates/instructor/instructor_dashboard_2/data_download.html:
```diff
@@ -1,9 +1,33 @@
+  padding: 60px;
+  line-height: 1.9;
+}
+</style>
```
It's ok to put CSS here or we need it compiled?

felipemontoya replied:
We need to put it in the static sass/css pipeline

dfrojas commented on lms/templates/instructor/instructor_dashboard_2/instructor_dashboard_2.html:
```diff
@@ -143,6 +144,18 @@ <h3 class="hd hd-3" id="header-${ section_data['section_key'] }">${ section_data
   </div>
 </div>
 
+% if configuration_helpers.get_value('CUSTOM_THEME_DISPLAY_SENSITIVE_DATA_MSG_FOR_DASHBOARD', settings.FEATURES.get('CUSTOM_THEME_DISPLAY_SENSITIVE_DATA_MSG_FOR_DASHBOARD', True)):
```
Was necessary place this code here due to JS in `data_download.html` executes in all tabs, or I'm missing something of how can I execute JS in a single tab once the page is loaded?

dfrojas commented on lms/templates/instructor/instructor_dashboard_2/instructor_dashboard_2.html:
```diff
@@ -143,6 +144,18 @@ <h3 class="hd hd-3" id="header-${ section_data['section_key'] }">${ section_data
+</script>
+
+% endif
+
```
Was necessary place this code here due to JS in `data_download.html` executes in all tabs, or I'm missing something of how can I execute JS in a single tab once the page is loaded?

dfrojas commented on lms/static/sass/course/instructor/_instructor_2.scss:
```diff
@@ -291,6 +291,18 @@
+  display: block;
+  z-index: 2;
+}
+
```
We could name `modal_close` with a more specific name for the data disclaimer modal, but this name it's more generic and could be useful to close any modal.

dfrojas commented on lms/templates/instructor/instructor_dashboard_2/data_download.html:
```diff
@@ -1,9 +1,35 @@
+      $( "#sensitive_data" ).click();
+  });
+});
+
```
Is not necessary write `function (require) {...` as we talk before, since the `<%static...` tag place it by itself, if you go to source code in the browser you'll see this:

```
<script type="text/javascript">
  (function (require) {
  (function (require) {
  require(['js/vendor/jquery.leanModal'], function (leanModal) { 
  $( document ).ready(function() {
  $('[data-section="data_download"]').click(function(){
  $( 'a[rel*=leanModal]' ).leanModal({ top : 200, closeButton: ".modal_close" });
  $( "#sensitive_data" ).click();
  });
  });  
  });
  }).call(this, require \|\| RequireJS.require);
  }).call(this, require \|\| RequireJS.require);
  </script>
```
But if you note, ` (function (require) {...`  appears twice, same happens with other function already configured:

```
<%static:require_module_async module_name="js/dateutil_factory" class_name="DateUtilFactory">
    DateUtilFactory.transform(iterationKey=".localized-datetime");
</%static:require_module_async>
```
Which render this:

```
<script type="text/javascript">
  (function (require) {
  (function (require) {
  require(['js/dateutil_factory'], function (DateUtilFactory) {  
  DateUtilFactory.transform(iterationKey=".localized-datetime");  
  });
  }).call(this, require \|\| RequireJS.require);
  }).call(this, require \|\| RequireJS.require);
  </script>
```



It is normal or I'm missing something?

dfrojas commented on lms/templates/instructor/instructor_dashboard_2/data_download.html:
```diff
@@ -1,9 +1,35 @@
+<%static:require_module_async module_name="js/vendor/jquery.leanModal" class_name="leanModal">
+
+$( document ).ready(function() {
+  $('[data-section="data_download"]').click(function(){
```
The modal is displayed only when the user does click on the tab, but if the user goes directly to the URL `instructor#view-data_download` it does not display the modal. This happens because if I place the code to display the modal after `$( document ).ready(function() {...`  it loads an opacity in all tabs, when I say an opacity, I mean this:

![screenshot from 2018-01-26 11-28-49](https://user-images.githubusercontent.com/3374009/35449847-27b4ff4c-028c-11e8-845d-d5d922a023fe.png)

Note that only loads the opacity but not the modal. 

I'm working on it right now, but question is: It is ok to display the modal only with click event or it's mandatory too when the DOM loads?

felipemontoya commented on lms/templates/instructor/instructor_dashboard_2/data_download.html:
```diff
@@ -1,9 +1,35 @@
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
 %>
 
+% if configuration_helpers.get_value('ENABLE_SENSITIVE_DATA_MSG_ON_DASHBOARD', settings.FEATURES.get('ENABLE_SENSITIVE_DATA_MSG_ON_DASHBOARD', True)):
```
I would leave the default as FALSE

felipemontoya commented on lms/templates/instructor/instructor_dashboard_2/data_download.html:
```diff
@@ -1,9 +1,35 @@
+
+<%static:require_module_async module_name="js/vendor/jquery.leanModal" class_name="leanModal">
+
+$( document ).ready(function() {
```
we can use the require pattern instead of waiting for the `jquery.ready` thing

felipemontoya replied:
https://github.com/edx/edx-platform/blob/7a69d2f771fbb89f2d2420811ca12578a5f2163f/cms/templates/widgets/source-edit.html#L31

jagonzalr commented:
@dfrojas can we add the translation strings also in the english folder (if they are not there yet)?

dfrojas commented:
@jagonzalr I just added the string in English folder. 

We have added the .po files just for your testing purposes, it's possible that for merge in upstream, we have to remove these files.

dfrojas commented:
@jagonzalr We have added the .po files just for your testing purposes, it's possible that for merge in upstream, we have to remove these files.

We have added the string in the .po English file, the diff is very big when I extracted the strings because these fields are not extracted often. But as I told you, for merge in upstream, it's possible that these files will be removed
